docs/process/release.md

```diff
@@ -47,9 +48,9 @@ Once the release checklist is complete, the assigned release driver will perform
     * `cargo owner -a github:unicode-org:icu4x-release`
 * Merge the tutorials PR. `cargo make test-tuturials-cratesio` should now pass
   * If there are any errors, please fix them before advertising the release
-* [Tag the Release](https://github.com/unicode-org/icu4x/releases)
 * Land the changelog (see below)
 * Announce the release to public
+  * [Tag the Release](https://github.com/unicode-org/icu4x/releases) with the text drafted above
```

Comment:
I think we should tag the release first, and then publish from that tagged release.

Reply:
This is not practical because we discover problems that need to be fixed after we've started running publish commands and there's no way to do a smoke test